server/src/com/gip/xyna/xprc/xfractwfe/python/PythonMdmGeneration.java

```diff
@@ -0,0 +1,493 @@
+    XynaObjectInformation result = new XynaObjectInformation();
+    result.fqn = fqn;
+    try {
+      if (isException) {
+        ExceptionGeneration obj = ExceptionGeneration.getOrCreateInstance(fqn, cache, revision);
+        obj.parse(false);
+        result.parent = obj.getSuperClassGenerationObject() != null ? obj.getSuperClassGenerationObject().getOriginalFqName() : null;
+        result.members = obj.getMemberVars().stream().map(this::toMemberInfo).collect(Collectors.toList());
+      } else {
+        DOM obj = DOM.getOrCreateInstance(fqn, cache, revision);
+        obj.parse(false);
+        result.parent = obj.getSuperClassGenerationObject() != null ? obj.getSuperClassGenerationObject().getOriginalFqName() : null;
+        result.members = obj.getMemberVars().stream().map(this::toMemberInfo).collect(Collectors.toList());
+        result.methods = loadOperations(obj.getOperations());
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    if (result.parent == null) {
+      result.parent = isException ? "XynaException" : "XynaObject";
+    } else {
+      result.parent = result.parent.replace('.', '_');
+    }
+    
+    //special case XynaException (inherits from java.lang.Exception
+    if (fqn.equals("core.exception.XynaException")) {
+      result.parent = "XynaException";
+    }
+
+    return result;
+  }
+
+
+  private Pair<String, String> toMemberInfo(AVariable avar) {
+    return new Pair<String, String>(avar.getVarName(), getPythonTypeOfVariable(avar));
+  }
+
+
+  private List<MethodInformation> loadOperations(List<Operation> operations) {
+    if (operations == null || operations.isEmpty()) {
+      return null;
+    }
+    List<MethodInformation> result = new ArrayList<MethodInformation>();
+    for (Operation op : operations) {
+      MethodInformation info = new MethodInformation();
+      info.isStatic = op.isStatic();
+      info.name = op.getNameWithoutVersion();
+      info.returnType = createReturnTypeFromOutputVars(op.getOutputVars());
+      info.argumentsWithTypes = createArgumentsWithTypes(op.getInputVars());
+      result.add(info);
+    }
+    return result;
+  }
+
+
+  private List<Pair<String, String>> createArgumentsWithTypes(List<AVariable> inputVars) {
+    List<Pair<String, String>> result = new ArrayList<Pair<String, String>>();
+    if (inputVars == null || inputVars.isEmpty()) {
+      return null;
+    }
+    for (AVariable avar : inputVars) {
+      result.add(new Pair<String, String>(avar.getVarName(), getPythonTypeOfVariable(avar)));
+    }
+    return result;
+  }
+
+
+  private String getPythonTypeOfVariable(AVariable avar) {
+    String type;
+    if (avar.isJavaBaseType()) {
+      type = primitive_types_mapping.getOrDefault(avar.getJavaTypeEnum(), "any");
+    } else {
+      
+      type = "'" +avar.getOriginalPath().replace('.', '_') + "_" + avar.getOriginalName() + "'";
```

Comment:
maybe we should define a method, witch converts javaFQN in Python classnames. So we can ensure every convert is equal.